test/jdk/java/lang/Class/getDeclaredField/FieldSetAccessibleTest.java

```diff
@@ -283,18 +288,64 @@ public Iterator<String> iterator() {
         }
 
         /*
-         * Filter deployment modules
+         * Filter deployment modules and JVMCI module and its transitive dependences
          */
         static Set<String> systemModules() {
-            Set<String> mods = Set.of("javafx.deploy", "jdk.deploy", "jdk.plugin", "jdk.javaws",
-                // All JVMCI packages other than jdk.vm.ci.services are dynamically
-                // exported to jdk.internal.vm.compiler
-                "jdk.internal.vm.compiler"
+            // Build module graph and inverse dependences
+            Set<String> modules = new HashSet<>();
+            Map<String, Set<String>> moduleToDeps = new HashMap<>();
+            Map<String, Set<String>> inverseDeps = new HashMap<>();
+            for (ModuleReference mref : ModuleFinder.ofSystem().findAll()) {
+                var md = mref.descriptor();
+                modules.add(md.name());
+                Set<String> deps = md.requires().stream().map(ModuleDescriptor.Requires::name)
+                                                .collect(Collectors.toSet());
+                moduleToDeps.put(md.name(), deps);
+                inverseDeps.put(md.name(), new HashSet<>());
+            }
+
+            // reverse edges
+            moduleToDeps.keySet().forEach(u -> {
+                moduleToDeps.get(u)
+                            .forEach(v -> inverseDeps.get(v)
+                                                     .add(u));
+            });
+
+            Set<String> deployModules = Set.of("javafx.deploy", "jdk.deploy", "jdk.plugin", "jdk.javaws");
```

Comment:
Can deployModules be dropped?

Reply:
There are two other tests that should be updated.  I updated this one in this PR.  Will create another PR to drop those.

Reply:
Okay, that's fine.